lib/hocon/impl/simple_config_origin.rb

```diff
@@ -25,23 +28,173 @@ def initialize(description, line_number, end_line_number,
     @comments_or_nil = comments_or_nil
   end
 
-  attr_reader :description, :line_number, :end_line_number, :origin_type,
+  attr_reader :_description, :line_number, :end_line_number, :origin_type,
               :url_or_nil, :resource_or_nil, :comments_or_nil
 
 
   def self.new_simple(description)
     self.new(description, -1, -1,
-             Hocon::Impl::OriginType::GENERIC,
+             OriginType::GENERIC,
              nil, nil, nil)
   end
 
   def self.new_file(file_path)
     url = URI.join('file:///', file_path)
     self.new(file_path, -1, -1,
-             Hocon::Impl::OriginType::FILE,
+             OriginType::FILE,
              url, nil, nil)
   end
 
+  def self.new_url(url)
+    u = url.to_external_form
```

Comment:
From what I can tell reading the java docs, `toExternalForm` the same as `URI#to_s`, and I've used that successfully in a couple tests